[ juneja999 ] ip

juneja999 opened the pull request:
Added capabilities of greeting, listing and marking/unmarking tasks

Commit:
Change name of Bot to AJ

Commit:
Add echo functionality of bot

The bot echoes user command and exits when commanded with "bye"

Commit:
Add storing and listing user commands

Commit:
Add marking and unmarking capabilities for tasks

Commit:
Improve coding standard

wongyihao02 commented on src/main/java/AJ.java:
```diff
@@ -0,0 +1,79 @@
+
+                for(int i = 0; i < currentList.length; i++){
+                    System.out.print(" ".repeat(AJ_TEXT_INDENTATION)+ (i+1)+"] "+currentList[i]); //because indexing starts from 0
+                    System.out.println(" ".repeat(AJ_TEXT_INDENTATION)+(taskList.taskList.get(i).isDone ? "{completed}" :""));//the taskList has a protected attribute taskList(mention in file TaskList, I'm accesing the index of that attribute)
```
Might be good to standardize putting comments on the line before or after the line it refers to.Putting all on the same line might lead to the line being too long with long comments,not so good for smaller screens.

juneja999 replied:
Yes, I also feel the same. I will improve the structure in upcoming commits.

wongyihao02 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,34 @@
+    protected ArrayList<Task> taskList = new ArrayList<Task>();
+
+    public TaskList() {}
+    public TaskList(Task task) {
```
is this constructor needed?TaskList is instantiated empty in main.

wongyihao02 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,14 @@
+    protected Boolean isDone;
+
+    //this is so that we can get the Name instead of memory address when converting Task to String
+    public String getTaskDescription(Task task){
```
maybe you could use a toString method?no need to put in a Task obj,just directly ask the obj what the taskDescription is.Or can be used as a way to have a standardized way to print out tasks.

juneja999 replied:
This has been kept from an earlier version, now its not being used anywhere in the code. I will try to phase this out in the following commits.

wongyihao02 reviewed (commented):
Overall,i like the personality that has been added to your chatbot.

Commit:
Refactor code

Extracted variables from LoC and changed the if-else system with a mix
of if-else and switch-cases for more readable code

Commit:
Fix for when extra whitespaces inputted

prathisttam commented on src/main/java/AJ.java:
```diff
@@ -0,0 +1,104 @@
+
+public class AJ {
+
+    private final static String DOTTED_LINE="***********************(^‿^)***************************";
```
Good job separating out the constant line to reuse as a constant!

juneja999 replied:
Thanks

prathisttam commented on src/main/java/AJ.java:
```diff
@@ -0,0 +1,104 @@
+public class AJ {
+
+    private final static String DOTTED_LINE="***********************(^‿^)***************************";
+    private final static String AJ_TEXT_INDENTATION = " ".repeat(7);
```
I really like the use of the method repeat().  Use of this makes it easy to modify the code in the future rather than retyping out the string literal multiple times.

prathisttam commented on src/main/java/AJ.java:
```diff
@@ -0,0 +1,104 @@
+                //to tackle edge cases - when extra white spaces between words
+                String targetWord = userCommandSplitArray[0].toLowerCase();
+
+                switch (targetWord){
```
Good use of "Switch Case" to prevent arrowhead nested if-else statements.

prathisttam commented on src/main/java/AJ.java:
```diff
@@ -0,0 +1,104 @@
+                String[] userCommandSplitArray = userCommandParser(userCommand.split(" "));
+                // array containing only user words, no whitespaces!
+                //to tackle edge cases - when extra white spaces between words
+                String targetWord = userCommandSplitArray[0].toLowerCase();
```
Should naming convention for targetWord be clearer to indicate what exactly targetWord means?

prathisttam reviewed (commented):
Overall, your code looks good ! You could consider adding more Javadoc comments throughout your code and splitting the main java code into additional classes for better modularity.

Commit:
Add capabilities for Todos, Deadlines and Events

Commit:
Improve bot message after adding task

Commit:
Handle case when user doesn't use any target word

Commit:
Handle case when user doesn't use any target word

Commit:
Add error handling mechanism for marking/unmark non-number literals

Commit:
Handle error for index out of bounds for mark/unmark

Commit:
Merge branch 'branch-Level-5' of https://github.com/juneja999/ip into branch-Level-5

Commit:
Merge branch 'branch-Level-5'

irving11119 commented on src/main/java/AJ.java:
```diff
@@ -0,0 +1,172 @@
+
+public class AJ {
+
+    public final static String DOTTED_LINE="***********************(^‿^)***************************";
```
Take note of the spaces around the `=`operator

irving11119 commented on src/main/java/AJ.java:
```diff
@@ -0,0 +1,172 @@
+    public static String[] userCommandParser(String[] command){
+        ArrayList<String> output= new ArrayList<String>();
+        for(int i=0; i<command.length; i++){
+            if(command[i].equals("")){continue;}
```
Take note that the coding standard requires K&R style brackets. Your brackets should not be on the same line.

irving11119 commented on src/main/java/AJ.java:
```diff
@@ -0,0 +1,172 @@
+//                    //the taskList has a protected attribute taskList(mention in file TaskList,
+//                    // I'm accessing the index of that attribute)
+//
+//                }
```
Do note you should remove commented out code

irving11119 commented on src/main/java/AJ.java:
```diff
@@ -0,0 +1,172 @@
+                        }else{
+                            System.out.println("Sorry but that serial number was not recognised, please try again!");
+                        }
+                    }
```
Take note for readability, you should try to keep nesting to a maximum level of 3. You can do this by abstracting out more logic into methods

irving11119 commented on src/main/java/AJ.java:
```diff
@@ -0,0 +1,172 @@
+                        defaultCaseTurn = 0;
+
+                    }
+                    break;
```
There is no need for a break in the default case

irving11119 commented on src/main/java/Events.java:
```diff
@@ -0,0 +1,33 @@
+                if(flag1 == 0) {flag1 = 1;}else {flag2 = 1;}
+                continue; // <---------------------------------TO NOT HAVE MULTIPLE FROM/ TO
+            }
+            if(flag1==0 && flag2==0) {eventsData[0] += command[i]+" ";}
+            if( flag1==1 && flag2==0) {eventsData[1] += " " + (command[i].startsWith("/") ? command[i].substring(1) : command[i]);}
+            if( flag1==1 && flag2==1) {eventsData[2] += " " + (command[i].startsWith("/") ? command[i].substring(1) : command[i]);}
```
These conditional statements are quite complex. It is ok to have logic split across multiple lines if it helps to make your code more readable,

irving11119 commented on src/main/java/AJ.java:
```diff
@@ -0,0 +1,172 @@
+                if(defaultCaseTurn==1){targetWord="make sure user doesnt put in mark/unmark etc at this stage";}
+
+                switch (targetWord.toLowerCase()){ //<------feature
+                case "mark":
```
It is good practice to comment fall-through reasons in your switch case to explicitly show that the cases are handled by the same logic.

irving11119 commented on src/main/java/AJ.java:
```diff
@@ -0,0 +1,172 @@
+
+                default: //if user doesn't mention any of  the above
+
+                    if(defaultCaseTurn == 0){
+                        AJExceptions.taskTypeNotFound(userCommand);
+                        ToDos tempDefaultCase = new ToDos(userCommand);
+                        taskList.addTask(tempDefaultCase);
+                        defaultCaseTurn = 1;
+                    }else{
+                        if(userCommand.equals("yes")){
+                            System.out.println(AJ_TEXT_INDENTATION + "Done!, added: "+
+                                    taskList.taskList.get(taskList.getSize()-1).getTaskInfo());
+                            //taskList has its own taskList(from class TaskList)
+                        } else if (userCommand.equals("no")) {
+                            System.out.println(AJ_TEXT_INDENTATION + "Okay, got it boss!");
+                            taskList.removeTask(taskList.getSize()-1);
+                        }else{
+                            System.out.println(AJ_TEXT_INDENTATION + "That command could not be recognised " +
+                                    "for this instance." );
+                            System.out.println(AJ.AJ_TEXT_INDENTATION+"Please try adding the task again. Thanks!");
+                            taskList.removeTask(taskList.getSize()-1);
+                        }
+                        defaultCaseTurn = 0;
```
To improve readability in the code, you can group similar statements together and separate groups with whitespace lines. This can help to increase readability of your code.

irving11119 commented on src/main/java/Deadlines.java:
```diff
@@ -0,0 +1,36 @@
+        String deadlinesByDate="";
+        String deadlinesDescription="";
+        int flag=0;
+        for(int i=1;i<command.length;i++){ //first word is deadlines, so skip it
+            if(command[i].startsWith("/")){
+                //deadlinesByDate+=command[i].substring(1); <------- TO NOT HAVE MULTIPLE BY BY: STATEMENTS
+                flag=1;
+                continue;
+            }
+            if(flag==1){
+                deadlinesByDate+=" "+command[i];
+
+            }else{
+                deadlinesDescription+=command[i]+" ";
+            }
+        }
```
Take note of coding standards regarding spaces around brackets, conditional statements and `==` operators

irving11119 commented on src/main/java/Events.java:
```diff
@@ -0,0 +1,33 @@
+public class Events extends Task {
+    public String From;
+    public String To;
```
Take note that variable names should be in camelCase. This applies to arguments supplied to methods as well.

irving11119 commented on src/main/java/Events.java:
```diff
@@ -0,0 +1,33 @@
+
+    public static String[] getEventsData(String[] command) {
+        String[] eventsData = new String[]{"","",""};
+        int flag1 = 0;
+        int flag2 = 0;
```
Maybe you could use more descriptive variable naming for your flags. This could help code reviewers identify exactly what the flags are used for in your code.

irving11119 reviewed (commented):
Do note some of the coding standard violations. Additionally, for better readability of your code, you can try to abstract out your code to reduce nesting and organize statements together for better code readability.

Commit:
Create JAR file version 1

Commit:
Implement task deleting capabilities

Commit:
Saving tasks locally v1 (separate files)

Commit:
Save tasks locally v2

Commit:
Merge branch 'branch-level-6'

Commit:
Merge branch 'branch-level-7'

Commit:
Make code more OOP aligned

Commit:
Make code more OOP aligned v2.0

Commit:
Make code more OOP aligned v3.0

Commit:
Make code more OOP aligned v4.0

Commit:
Interpret dates and time

Commit:
Add Task finding capabilities

Commit:
Add JavaDoc comments to classes and methods

juneja999 committed:
Merge pull request #1 from juneja999/branch-level-8

Interpret dates and time

Commit:
Merge branch 'master' into branch-level-9

# Conflicts:
#	UserData/SavedTasks.txt

Commit:
Merge branch 'master' into branch-A-JavaDoc

juneja999 committed:
Merge pull request #2 from juneja999/branch-level-9

Add Task finding capabilities

Commit:
Merge branch 'master' into branch-A-JavaDoc

juneja999 committed:
Merge pull request #3 from juneja999/branch-A-JavaDoc

Add JavaDoc comments to classes and methods

Commit:
Add a sufficiently detailed user guide

juneja999 committed:
Update README.md

juneja999 committed:
Update README.md  v2.0

juneja999 committed:
Update README.md v3.0

juneja999 committed:
Update README.md v4.0

juneja999 committed:
Update README.md v5.0

Commit:
Change format of README.md

juneja999 committed:
Update README.md v6.0

juneja999 committed:
Update README.md v7.0

Commit:
Change format of README.md format v2.0

juneja999 committed:
Update README.md v8.0

Commit:
Enhance default task case

Commit:
Make NoCommand message

Commit:
Enhance the task finding method
